### PR TITLE
Live reload rework

### PR DESCRIPTION
**Pull request by digitalsadhu**
Reworks live reload based on @trygve-lie prototype here https://github.com/trygve-lie/prototype-build-server/blob/main/bin/plugin-live-reload.js

Essentially, this moves all client side live reload into a single plugin including file watching, web socket setup and use and client side scripts injected into the page. This simplifies the scripting required to start the server in dev mode. 

Subsequent PRs will refactor the server side file watching and loading.

### CODE REVIEW

**Comment by trygve-lie** [comment]
We should also a ping / pong feature in the web socket implementation to keep long running sessions alive. The initial prototype has one: https://github.com/trygve-lie/prototype-build-server/blob/main/bin/plugin-live-reload.js#L43-L68

File: plugins/live-reload.js [hunk 1]
```diff
@@ -1,8 +1,104 @@
+// @ts-ignore
+import chokidar from "chokidar";
 import fp from "fastify-plugin";
+import etag from "@fastify/etag";
+import cors from "@fastify/cors";
+import websocket from "@fastify/websocket";
```
[the remark this answers is not in the document]
Comment (digitalsadhu) [reply plugins/live-reload.js:6]: I debugged an issue with holding the process open and found that ws.close() was not finishing. Seemed like there were event listeners that needed to be removed. Looking at the code of @fastify/websocket I see that they do this book keeping and so it seemed that It was just cleaner/easier to use that. If you think we should avoid it I can see if I can unregister event handlers and get it to close down properly. Thoughts?

[the remark this answers is not in the document]
Comment (digitalsadhu) [reply plugins/live-reload.js]: Discovered the root of the issue:
@fastify/websocket uses 8.0.0 which works fine but the latest version hangs the process 8.13.0
I discovered this when i tried reverting the code back to raw websocket code but used the dep transiently through @fastify/websocket when i saw it worked, i installed ws directly and the process began hanging again
so i installed an older version of ws which matched the version being used in @fastify/websocket and everything worked again

**Comment by digitalsadhu** [comment]
@trygve-lie I've refactored back to use raw web sockets and included the ping pong code.

**Review by trygve-lie — APPROVED** [review]
lgtm :+1:

**Pull request merged**